main/src/http/header.rs

```diff
@@ -0,0 +1,160 @@
+use super::Error as HttpError;
+use spi::service::protocol::UserInfo;
+use warp::http::header::{HeaderName, HeaderValue};
+
+// re-export const header names
+pub use warp::http::header::{ACCEPT, AUTHORIZATION, CONTENT_TYPE};
+
+/// value
+pub const APPLICATION_PREFIX: &str = "application/";
+pub const APPLICATION_CSV: &str = "application/csv";
+pub const APPLICATION_TSV: &str = "application/tsv";
+pub const APPLICATION_JSON: &str = "application/json";
+pub const APPLICATION_NDJSON: &str = "application/nd-json";
+pub const APPLICATION_STAR: &str = "application/*";
+pub const STAR_STAR: &str = "*/*";
+
+/// basic auth
+const BASIC_PREFIX: &str = "Basic ";
+
+#[derive(Debug, Clone)]
+pub struct Header {
+    accept: Option<String>,
+    authorization: String,
+}
+
+impl Header {
+    pub fn with(accept: Option<String>, authorization: String) -> Self {
+        Self {
+            accept,
+            authorization,
+        }
+    }
+
+    pub fn get_accept(&self) -> &str {
+        self.accept.as_deref().unwrap_or(APPLICATION_CSV)
+    }
+
+    pub fn try_get_basic_auth(&self) -> Result<UserInfo, HttpError> {
+        let auth = &self.authorization;
+
+        let err = Err(HttpError::ParseAuth {
```

Comment:
How about using this?
```rust
let get_err = || Err(HttpError::ParseAuth {
    reason: auth.to_string(),
});
```

Reply:
The same exception is returned in several places, so initialization is performed at the beginning

Reply:
I agree

Reply:
Can I hear your thoughts